geth/jail/jail.go

```diff
@@ -109,60 +139,49 @@ func (jail *Jail) Parse(chatID string, js string) string {
 // Call executes the `call` function w/i a jail cell context identified by the chatID.
 // Jail cell is clonned before call is executed i.e. all calls execute w/i their own contexts.
 func (jail *Jail) Call(chatID string, path string, args string) string {
-	jail.RLock()
-	cell, ok := jail.cells[chatID]
-	if !ok {
-		jail.RUnlock()
-		return makeError(fmt.Sprintf("Cell[%s] doesn't exist.", chatID))
-	}
-	jail.RUnlock()
-
-	// Due to the new timer assigned we need to clone existing cell to allow
-	// unique cell runtime and eventloop context.
-	cellCopy, err := cell.Copy()
+	jcell, err := jail.getCell(chatID)
 	if err != nil {
 		return makeError(err.Error())
 	}
 
-	// isolate VM to allow concurrent access
-	vm := cellCopy.CellVM()
+	vm := jcell.VM()
+	loop := jcell.Loop()
+
+	jcell.Lock()
+	defer jcell.Unlock()
+
 	res, err := vm.Call("call", nil, path, args)
 
+	// WARNING(influx6): We can have go-routine leakage due to continous call to this method
+	// and the call to cell.CellLoop().Run() due to improper usage, let's keep this
+	// in sight if things ever go wrong here.
+	// Due to the new event loop provided by ottoext.
+	// We need to ensure that all possible calls to internal setIntervals/SetTimeouts/SetImmediate
+	// work by lunching the loop.Run() method.
+	// Needs to be done in a go-routine.
+	// TODO(influx6): Figure out if we should do a throw Exception for returned error
+	// from loop.Run() call.
+	go loop.Run()
+
 	return makeResult(res.String(), err)
 }
 
 // JailCellVM returns instance of Otto VM (which is persisted w/i jail cell) by chatID
+// TODO(influx6): Decide if this make sense now that we are guarding JailCell vm with mutex.
```

Comment:
Please, remove this method and fix tests. It's only used there.
Moreover, we shouldn't expose `vm` and `loop` to other parties because we need to guarantee that all calls are made in sequence so you'll have to additionally check that only JailCell methods are used, not its `vm` or `loop`.